packages/commerce-server/src/record-new-purchase.ts

```diff
@@ -126,23 +126,50 @@ export async function recordNewPurchase(checkoutSessionId: string): Promise<{
     stripeChargeAmount,
   })
 
-  if (purchase && quantity > 1) {
-    // we need to detect and handle the scenario where the Coupon already
-    // exists and we are just tying a new purchase to it. Eventually.
+  // Check if this user has already purchased a bulk coupon, in which case,
+  // we'll be able to treat this purchase as adding seats.
+  const existingBulkCoupon = await prisma.coupon.findFirst({
+    where: {
+      maxUses: {
+        gt: 1,
+      },
+      bulkCouponPurchases: {
+        some: {userId: user.id},
+      },
+    },
+  })
 
+  // Note: if the user already has a bulk purchase/coupon, then if they are
+  // only adding 1 seat to the team, then it is still a "bulk purchase" and
+  // we need to add it to their existing Bulk Coupon.
+  const isBulkPurchase = quantity > 1 || !!existingBulkCoupon
+
+  if (purchase && isBulkPurchase) {
```

Comment:
Here in the `record-new-purchase` function (which gets called by the Stripe webhook handler for the `checkout.session.completed` event), we can look up whether this user already has made a Bulk Coupon purchase. If they have, then whether they are adding `1` seat or `10` seats, we want to treat this as an _additional_ bulk purchase to tack on to the existing coupon.